layouts/partials/fragments/footer.html

```diff
@@ -1,10 +1,43 @@
-{{- $name := strings.TrimSuffix ".md" (replace .self.Name "/index.md" "") -}}
 {{- $bg := .Params.background | default "dark" }}
 
+{{/* Resource fallthrough path preparation */}}
+{{/* Do not change the following snippet */}}
+{{/* Code is duplicated throughout the code */}}
+{{- $path := .self.File.Path -}}
+{{- $path := replace $path "/index.md" "" -}}
+{{- $path := strings.TrimSuffix ".md" $path -}}
+{{- $page_path := replace $path (printf "/%s" .Name) "" -}}
+{{/* End of do not change */}}
```

Comment:
These two variables can be passes from the `single.html` layout to the fragments. There we have access to `.self` and `.Name` so there shouldn't be any problem and the code would not be duplicated then.

Reply:
Good point. Thanks for pointing out.